apps/marketplace/components/shared/form/FilesInput.js

```diff
@@ -0,0 +1,77 @@
+/* global FormData */
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { actions } from 'react-redux-form'
+import range from 'lodash/range'
+import dmapi from '../../../services/apiClient'
+
+import FileInput from './FileInput'
+
+class FilesInput extends React.Component {
+  static propTypes = {
+    name: PropTypes.string.isRequired,
+    label: PropTypes.string.isRequired,
+    hint: PropTypes.string.isRequired,
+    description: PropTypes.string.isRequired,
+    url: PropTypes.string.isRequired,
+    model: PropTypes.string.isRequired,
+    formFields: PropTypes.number.isRequired
+  }
+
+  static defaultProps = {}
+
+  render() {
+    const { label, description, hint, formFields } = this.props
+
+    return (
+      <div>
+        <label className="question-heading uikit-text-input__label" htmlFor="file_0">
+          {label}
+        </label>
+        <span>
+          {description}
+        </span>
+        <p className="hint">
+          {hint}
+        </p>
+        {range(formFields).map((field, id) => <FileInput key={field} id={id} {...this.props} />)}
+      </div>
+    )
+  }
+}
+
+const uploadDocument = (id, url, file) => () => {
+  const data = new FormData()
+  data.append(id, file)
+
+  return dmapi({
```

Comment:
@matt-sm just did an auth PR, isn't there session data? also, this might be better in a middleware?